projects/alembic/alembic_dump_info_fuzzer.cc

```diff
@@ -0,0 +1,203 @@
+using Alembic::AbcGeom::M44f;
+using Alembic::AbcGeom::N3fArraySamplePtr;
+using Alembic::AbcGeom::ObjectHeader;
+using Alembic::AbcGeom::P3fArraySamplePtr;
+using Alembic::AbcGeom::UInt32ArraySamplePtr;
+using Alembic::AbcGeom::V2fArraySamplePtr;
+using Alembic::AbcMaterial::IMaterial;
+using Alembic::AbcMaterial::IMaterialSchema;
+
+template <typename T> void dumpAttributes(T const &schema) {
+
+  const size_t meshPropertyCount = schema.getNumProperties();
+
+  for (size_t p = 0; p < meshPropertyCount; p++) {
+    const PropertyHeader &header = schema.getPropertyHeader(p);
+    const PropertyType pType = header.getPropertyType();
+    const std::string &name = header.getName();
+
+    if (name == "P") {
+      schema.getNumSamples();
+    } else if (name == "uv" || name == "st") {
+      schema.getUVsParam().getNumSamples();
+    } else if (name == ".arbGeomParams") {
+      // additional geometry elements (color sets, additional texture
+      // coordinates)
+      const ICompoundProperty geoParam = schema.getArbGeomParams();
+      const size_t geoPropCount = geoParam.getNumProperties();
+
+      for (size_t g = 0; g < geoPropCount; g++) {
+        const PropertyHeader &headerGeo = geoParam.getPropertyHeader(g);
+        const std::string &nameGeo = headerGeo.getName();
+      }
+    }
+  }
+}
+
+void dumpPolyMesh(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IPolyMesh mesh(node.getParent(), header.getName());
+  const IPolyMeshSchema &schema = mesh.getSchema();
+
+  // Mesh properties
+  dumpAttributes(schema);
+}
+
+void dumpSubD(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const ISubD mesh(node.getParent(), header.getName());
+  const ISubDSchema &schema = mesh.getSchema();
+
+  dumpAttributes(schema);
+  schema.getSubdivisionSchemeProperty();
+  schema.getFaceVaryingInterpolateBoundaryProperty();
+  schema.getFaceVaryingPropagateCornersProperty();
+  schema.getInterpolateBoundaryProperty();
+}
+
+void dumpFaceSet(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IFaceSet faceSet(node.getParent(), header.getName());
+  const IFaceSetSchema &schema = faceSet.getSchema();
+  schema.getNumSamples();
+}
+
+void dumpCurves(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const ICurves curves(node.getParent(), header.getName());
+  const ICurvesSchema &schema = curves.getSchema();
+
+  dumpAttributes(schema);
+}
+
+void dumpXform(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IXform xform(node.getParent(), header.getName());
+  const IXformSchema &schema = xform.getSchema();
+
+  schema.getNumSamples();
+  schema.getNumOps();
+}
+
+void dumpMaterial(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  IMaterial material(node.getParent(), header.getName());
+  IMaterialSchema &schema = material.getSchema();
+
+  std::vector<std::string> targetNames;
```

Comment:
I think you're missing the call to `IMaterialSchema.GetTargetNames` ([here](https://github.com/alembic/alembic/blob/master/lib/Alembic/AbcMaterial/IMaterial.h#L107)) to actually fill `targetNames` with the values from `schema`.

Reply:
Good catch!